### PR TITLE
chore: upgrade subtractions to v6 api

### PR DESCRIPTION
**Pull request by jasont7**
## Pre-Review Checklist
* [x] I have considered backwards and forwards compatibility.
* [x] All changes are tested.
* [x] All touched code documentation is updated.
* [x] All tests pass in my local environment
* [x] Deepsource issues have been reviewed and addressed.
* [x] Commented code and `console` usages have been removed.
 
## Screenshots
_Only required for visual changes_

### CODE REVIEW

**Comment by jasont7** [comment]
Not ready to merge - still investigating `SubtractionDetail` test failure

File: src/js/subtraction/components/Subtraction.tsx [hunk 1]
```diff
@@ -12,11 +12,11 @@ export default function Subtraction() {
...
-                <Switch>
-                    <Route path="/subtractions" component={SubtractionList} exact />
-                    <Route path="/subtractions/files" component={SubtractionFileManager} />
-                    <Route path="/subtractions/:subtractionId" component={SubtractionDetail} />
-                </Switch>
+                <Routes>
+                    <Route path="/subtractions" element={<SubtractionList />} />
+                    <Route path="/subtractions/files" element={<SubtractionFileManager />} />
+                    <Route path="/subtractions/:subtractionId" element={<SubtractionDetail />} />
+                </Routes>
```
Comment (ReeceHoffmann) [inline src/js/subtraction/components/Subtraction.tsx]: On a glance, I think the test failure is a path mismatch problem. Currently when directly rendering `Subtraction`, react router thinks that the URL of the details page is `/subtractions/:subtractionId` while in testing the URL being passed to MemoryRouter is `/:subtractionId`.

To resolve I would do two things:

1. In main update the `Route` used for subtratcions to `CompatRoute` with path `/subtractions`. (use samples as an example)
2. Update these paths to use the relative addressing

I think doing both should resolve the testing problems

Comment (jasont7) [reply src/js/subtraction/components/Subtraction.tsx]: Good catch!

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**